papers/libceed-paper/paper.rst

```diff
@@ -0,0 +1,72 @@
+
+.. figure:: TensorVsAssembly.png
+   :align: center
+   :figclass: bht
+
+   Comparison of memory transfer and floating point operations per degree of freedom for different representations of a linear operator for a PDE in 3D with :math:`b` components and variable coefficients arising due to Newton linearization of a material nonlinearity. The representation labeled as *tensor* computes metric terms on the fly and stores a compact representation of the linearization at quadrature points. The representation labeled as *tensor-qstore* pulls the metric terms into the stored representation. The *assembled* representation uses a (block) CSR format.
+
+In this work, we first introduce libCEED’s conceptual framework and C interface, and then illustrate its new Python interface, developed using the C Foreign Function Interface (CFFI) for Python. CFFI allows to reuse most of the C declarations and requires only a minimal adaptation of some of them. The C and Python APIs are mapped in a nearly 1:1 correspondence. For instance, data stored in the CeedVector structure are associated to arrays defined via the numpy package. In fact, since libCEED heavily relies on pointers and arrays to handle the data, a Python structure that resembles the C arrays is needed. In details, numpy arrays allow this correspondence obtained by passing the numpy array memory address as pointers to the libCEED C API.
+
+libCEED's API
+----------------------------------------------------------------------------------
+
+LibCEED's API provides the local action of the linear or nonlinear operator without assembling its sparse representation. Let us define the global operator as
+
+.. math::
+   :label: eq-operator-decomposition
+
+   A = P^T \underbrace{G^T B^T D B G}_{\text{libCEED}} P \, ,
+
+where :math:`P` is the parallel process decomposition operator (external to ``libCEED``) in which the degrees of freedom (DOFs) are scattered to and gathered from the different compute devices. The operator denoted by :math:`A_L = G^T B^T D B G` gives the local action on a compute node or process, where :math:`G` is a local element restriction operation that localizes DOFs based on the elements, :math:`B` defines the action of the basis functions (or their gradients) on the nodes, and :math:`D` is the user-defined pointwise function describing the physics of the problem at the quadrature points, also called the QFunction (see Fig. :ref:`fig-operator-decomp`).
```

Comment:
Perhaps here it would be appropriate to mention "This parallel decomposition is out of scope for the libCEED API and needs to be managed by the user or an additional package, such as petsc4py."

Reply:
We say (external to ``libCEED``) but sure

Reply:
Right, I mostly was thinking that we could highlight how libCEED can plug in with other packages that are already popular.